src/Controller/Component/CsrfComponent.php

```diff
@@ -56,7 +56,7 @@ class CsrfComponent extends Component
         'cookieName' => 'csrfToken',
         'expiry' => 0,
         'secure' => false,
-        'httpOnly' => false,
+        'httpOnly' => true,
```

Comment:
Found this: http://security.stackexchange.com/questions/59470/double-submit-cookies-vulnerabilities

In short, if your page is vulnerable to XSS, then basically no CSRF protection can be enforced... and think I missed before how obvious that is... If you can run arbitrary code in the same origin, there is no use for reading the cookies or trying to bypass CSRF protection at all, you are already in and can submit whatever you like.